### DISCUSSION TITLE
kotlin-spark-api in Apache Spark Codebase

### DISCUSSION THREAD

**Discussion by corzani** [JetBrains/kotlin-spark-api#209] [General]
Hi,
is there any update regarding Kotlin Spark being in the Apache Spark 3.x?
I would love to use this implementation to port our projects to Spark but it doesn't seem very active lately so I thought you might have paused it. Is this the case?

**Comment by Jolanrensen** [comment]
No update yet regarding Spark 3.4.x+, if that is what you mean by the question. Follow [the issue](https://github.com/Kotlin/kotlin-spark-api/issues/195) for updates if there are any.
Spark 3.0.0-3.3.2 is supported.

If you're curious about Kotlin support in the Apache Spark codebase, we have an ongoing Jira ticket you can follow:
https://issues.apache.org/jira/browse/SPARK-32530?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=17171633#comment-17171633. Unfortunately, the Spark team will likely not include Kotlin support in the foreseeable future. Just like .NET support it will probably remain a separate project.
